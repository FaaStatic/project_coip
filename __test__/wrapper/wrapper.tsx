import { RealmProvider } from '@realm/react';
import { QueryClientProvider } from '@tanstack/react-query';
import { KeyboardProvider } from 'react-native-keyboard-controller';
import { SafeAreaProvider } from 'react-native-safe-area-context';

export const Wrapper = ({ children, realm, queryClient }) => (
  <RealmProvider realm={realm}>
    <QueryClientProvider client={queryClient}>
      <SafeAreaProvider
        initialMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <KeyboardProvider>{children}</KeyboardProvider>
      </SafeAreaProvider>
    </QueryClientProvider>
  </RealmProvider>
);
