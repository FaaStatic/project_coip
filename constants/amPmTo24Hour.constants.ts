export const AmPmTo24Hours = {
  '12AM': '00',
  '01AM': '01',
  '02AM': '02',
  '03AM': '03',
  '04AM': '04',
  '05AM': '05',
  '06AM': '06',
  '07AM': '07',
  '08AM': '08',
  '09AM': '09',
  '10AM': '10',
  '11AM': '11',
  '12PM': '12',
  '01PM': '13',
  '02PM': '14',
  '03PM': '15',
  '04PM': '16',
  '05PM': '17',
  '06PM': '18',
  '07PM': '19',
  '08PM': '20',
  '09PM': '21',
  '10PM': '22',
  '11PM': '23',
};
export const HoursToAmPm = {
  '00': '12 AM',
  '01': '01 AM',
  '02': '02 AM',
  '03': '03 AM',
  '04': '04 AM',
  '05': '05 AM',
  '06': '06 AM',
  '07': '07 AM',
  '08': '08 AM',
  '09': '09 AM',
  '10': '10 AM',
  '11': '11 AM',
  '12': '12 PM',
  '13': '01 PM',
  '14': '02 PM',
  '15': '03 PM',
  '16': '04 PM',
  '17': '05 PM',
  '18': '06 PM',
  '19': '07 PM',
  '20': '08 PM',
  '21': '09 PM',
  '22': '10 PM',
  '23': '11 PM',
};
