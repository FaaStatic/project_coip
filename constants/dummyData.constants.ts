export const LIST_SIZE = 10;

export const dummyBenefit = [
  {
    title: 'Analisa Bahan Bakar',
    subtitle: 'Dapatkan rekomendasi penggunaan bahan kabar lebih detail & efektif',
    img: require('@untr/apps-coip/assets/images/analysis_img.png'),
  },
  {
    title: 'Dashboard Monitoring',
    subtitle: 'Pantau informasi terkait unit dan service ada melalui dashboard',
    img: require('@untr/apps-coip/assets/images/laptop_img.png'),
  },
  {
    title: 'Pelihara Alat Beratmu',
    subtitle: 'Gunakan fitur Maintenance Management untuk optimalkan kinerja alat berat',
    img: require('@untr/apps-coip/assets/images/truck_img.png'),
  },
];

export const base64Header = 'data:image/jpeg;base64,';

export const poinIcoBase64 =
  'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACYAAAAmCAYAAACoPemuAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAAGXRFWHRTb2Z0d2FyZQB3d3cuaW5rc2NhcGUub3Jnm+48GgAABTVJREFUWIXNmF1sVEUUx3/n7r0VRAot7baUauQb/EANvAj4gf1CDBGM2UiLS/DJz5j4YvTFSIxfCTFqeODJmLJiVo3GmIJboFYENIoPKohBig8V6LZABRTo3jvHByiU3blbdgvRf3If9pwzZ373zsyZ2RFV5f8o578GCNP/FswttmH34lh5ie/dj8h80ElA9LwrjdIN7LpmFG3jvkgcLya/FDrH0vXNdyLyElDH8C/mg2xReLmqfcO3VwWsr3FFjVFnHbCskA7OS1H9zHEjT1Vsbj18xcB6G1bcoTifATcUATVUh9QxD1V9ufG7EYOl65sbEPkcGDVCqEGdAZZG2xNbigbru+/RmSZidgFlVwhqUCdMoPOrt32wp2CwvbFYScVx7ydg5jCd9CmyTUS7AVSlVtA6YEK+Rgr7jpZlbrspmRyw+UNX1YTjJU+C5oM6qMqLVeWZj0gmg0s8sVikp9+NiZFXEW60NRaYNaHfewJ42+q3fbFjS1aW+hntIvytUx6ZWFl78q884PQvWj1+wBtIojSEhPS5nkwtb9twItthrfx+RpflgfoxGH3dchtUX+OKmiNLYpWDv8d3vNc/cCrzoMD3Ibkq/AFdanOEbUnLQ+yBCTQ+8fP1/1ig5hl19jkZb/vRulWTBu21O5OnfYJVgG/NKPa+csFEBKi35+Dj7JXU0xQfk25sWWvU2YhqHJXvA8fvPNyw6kLNm9j+4a/Ap1YwaLwssD/r4uXAdbZgo3JJ8iP3Nd8sJtgthjMqvCeOzDTGvK6wKaJ+Z09TfPJgrIqGgY3tXhwrzzbmrEqPzCQQawaFn4f+Fs8VTFBNkHlNxDupgBOR5qh/aG6vV9OtJujsbWypq0wl9qPOL2AvTZ6OqgGODbXZ5tiYkDejJOCS1VOVat0DnFHPHbp/zunzqm+vTCXeEJG3Vfmqt7FleuAE/WF5JfDHZttywBzM0bAEmZLgwqQ+2tQ8K13f/DswDpV1Q+P8wDmXV7USqDHo6oiJTCJE4kZ6s205Q1kR9HSl3Zq/sX05lVuB7wAGjPdPBH83sDs7zLhm8ARRAiBGShGdE8J1qrL07MFsY+5QdnT4wC5bBlHiFxsGDwFzbY9rnM09TfFbgNEKnWjwFlxsm6UdOTuHFQxA5BN7DlmYbmg5X0p0DjAl5JntBH5ttD3xRFV74l4i7lRggS2jin5s7cm2JfUvWj1+wB3oxjacQjdG50X7xx5Ll5263pY0YtyzE7a+/ydA76JYtbruDyC2OXbS9aTWtiWFni56GlreFXja6oQuERZXphL7Q/wA9NU/MsNIZBPnvmKOVHinKpV41uYLB2uKR8UE+4HSkH7/RmWt4wTrK1IbD10C1LiixuA8jvIc4eXnL+Nlple3JXNWZF4wgN76lS+o6KuhAeekCHtV5Q8AQScDswmr0hdayfPRLRveDHPnBeueHxtdcq23N+xMNQJ1nfCO3zStre1sWEDeP7y1O5OnBScO5CznEcg4oo/lgxoWDKByS+t2hOGGsxCtqUh90Dlc0GVdEUQzh9YAO0aMhG6PlmVeuZzIy7u76OjwA1+XAb+NgOqAOu7DtipvU0FXBOmGlinATqCqQKheERYMV/eGqqDbnmh7ostxeADoKwgK01QIFBRxqQKQrntkKk5kMzAtb6Dyh2OcxRXbWgueAkXdj0W3fnjAiTh3A9/kgfpagsydxUAVDQZQsbn1cHThjHsEngFODnGdQPWp6F0zFlV2JI8Um7+ooczWsSUrSwNfmzCikRJSttPCfwJ2NfQv45sUH6sDs/8AAAAASUVORK5CYII=';
