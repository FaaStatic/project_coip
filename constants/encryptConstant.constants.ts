const enryptConstant = {
  encryptionKey: new Int8Array(64),
};

export const { encryptionKey } = enryptConstant;
