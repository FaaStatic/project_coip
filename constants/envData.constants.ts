import { API_URL, OCP_APIM_KEY, JOB_CREATION_SWAGGER_URL, LOGIN_SWAGGER_URL } from '@env';

export const envData = {
  BASE_URL: API_URL ?? process.env.API_URL,
  APIM_KEY: OCP_APIM_KEY ?? process.env.OCP_APIM_KEY,
  JOB_CREATION_SWAGGER_URL: JOB_CREATION_SWAGGER_URL ?? process.env.JOB_CREATION_SWAGGER_URL,
  LOGIN_SWAGGER_URL: LOGIN_SWAGGER_URL ?? process.env.LOGIN_SWAGGER_URL,
};
