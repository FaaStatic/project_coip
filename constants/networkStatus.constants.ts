export enum STATUS_CODE {
  OK = 200,
  MOVED_PERMANENTLY = 301,
  BAD_REQUEST = 400,
  UNAUTHORIZED = 401,
  FORBIDDEN = 403,
  NOT_FOUND = 404,
  METHOD_NOT_ALLOWED = 405,
  INTERNAL_SERVER_ERROR = 500,
  BAD_GATEWAY = 502,
  SERVICE_UNAVAIBLE = 503,
}
