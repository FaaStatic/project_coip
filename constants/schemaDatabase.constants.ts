export enum schemaDatabase {
  Login = 'Login',
  CreateJobsItem = 'CreateJobsItem',
  CheckSheetJob = 'CheckSheetJob',
  Equipment = 'Equipment',
  Customer = 'Customer',
  Syncs = 'Syncs',
  CheckSheetMaster = 'CheckSheetMaster',
  FirstInstall = 'FirstInstall',
  CustomerList = 'CustomerList',
}
