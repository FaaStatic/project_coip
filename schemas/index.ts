import AdditionalJobs from './AdditionalJobs/AdditionalJobs.schema';
import CheckSheetJob from './CheckSheetJob/CheckSheetJob.schema';
import CheckSheetMaster from './CheckSheetMaster/CheckSheetMaster.schema';
import CreateJobsItem from './CreateJobsItem/CreateJobsItem.schema';
import Customer from './Customer/Customer.schema';
import JobChildSheet from './JobChildSheet/JobChildSheet.schema';
import FirstInstall from './FirstInstall/FirstInstall.schema';
import Equipment from './Equipment/Equipment.schema';
import EquipmentIdentity from './EquipmentIdentity/EquipmentIdentity.schema';
import Syncs from './Syncs/Syncs.schema';
import Login from './Login/Login.schema';
import CustomerList from './CustomerList/CustomerList.schema';

export default {
  AdditionalJobs,
  CheckSheetJob,
  CheckSheetMaster,
  CreateJobsItem,
  Customer,
  EquipmentIdentity,
  Equipment,
  FirstInstall,
  JobChildSheet,
  Login,
  CustomerList,
  Syncs,
};
