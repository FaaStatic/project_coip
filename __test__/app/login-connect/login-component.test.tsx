import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { describe, expect, test, jest, beforeEach } from '@jest/globals';
import { QueryClient, useMutation } from '@tanstack/react-query';
import LoginConnect from '@untr/apps-coip/app/login-connect/_layout';
import { Wrapper } from '../../wrapper/wrapper';
import Realm from 'realm';
import schemas from '@untr/apps-coip/schemas';
import { tanstackQueryMock } from '@untr/apps-coip/__mocks__/tanstack.mock';
import { routerMock } from '@untr/apps-coip/__mocks__/router.mock';
import { router } from 'expo-router';
import { zustandMock } from '@untr/apps-coip/__mocks__/zustand.mock';
import LoadingState from '@untr/apps-coip/states/loadingState.state';
import { TestUnitIDConstant } from '@untr/apps-coip/constants/testUnitID.constants';

const schemaVersion = 3;

const mockAddRealm = async () => {
  const configMockDB: Realm.Configuration = {
    schema: Object.values(schemas),
    schemaVersion,
    path: 'mockDb.realm',
    deleteRealmIfMigrationNeeded: true,
    inMemory: true,
  };

  try {
    await Realm.open(configMockDB);
  } catch (error) {
    console.log('Error MOCK DB', error);
  }
};

describe('Login Component test', () => {
  let realm;
  let queryClient;
  beforeEach(async () => {
    jest.clearAllMocks();
    realm = await mockAddRealm();
    queryClient = new QueryClient({
      defaultOptions: {
        queries: {
          // âœ… turns retries off
          retry: false,
          gcTime: Infinity,
        },
      },
    });
  });
  const mockMutate = tanstackQueryMock.fn().mockReturnValue({ useMutation });

  const routeMock = routerMock.fn().mockReturnValue({ router });
  const changeLoading = zustandMock.fn().mockReturnValue({ LoadingState });

  test('is Login Component Rendered?', async () => {
    const renderLogin = render(
      <Wrapper realm={realm} queryClient={queryClient}>
        <LoginConnect />
      </Wrapper>
    );
    await waitFor(() => {
      expect(renderLogin.toJSON()).toMatchSnapshot();
    });
  });

  test('Error show when loggin is Fail?', async () => {
    mockMutate.mockImplementationOnce(() => {
      throw { response: { data: { errors: { UserName: ['Invalid credentials'] } } } };
    });

    const { findByText, getByTestId, getByText, debug } = render(
      <Wrapper realm={realm} queryClient={queryClient}>
        <LoginConnect />
      </Wrapper>
    );

    debug();

    fireEvent.changeText(getByTestId(TestUnitIDConstant.LOGINCONNECT_FORM_USERNAME), 'testuser');
    fireEvent.changeText(getByTestId(TestUnitIDConstant.LOGINCONNECT_FORM_PASSWORD), 'password123');
    fireEvent.press(getByTestId(TestUnitIDConstant.LOGINCONNECT_BUTTON_LOGIN));

    await waitFor(() => findByText('Login Failed!'));
    expect(getByText('Login Failed!')).toBeTruthy();
    expect(getByText('Invalid credentials')).toBeTruthy();
  });

});
