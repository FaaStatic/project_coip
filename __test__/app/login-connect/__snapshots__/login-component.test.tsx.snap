// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component test is Login Component Rendered? 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
